From 3ed8e78d1277ceee388d7e1046f564835ec3a974 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jose.dapena@lge.com>
Date: Tue, 4 Jun 2019 19:00:36 +0200
Subject: GCC: remove noexcept in history::QueryURLResult move constructor and
 assign operator.

They declare the default move assign operator and constructor. But that requires all
members to declare theirs as noexcept.

Bug: 819294
Change-Id: I726e3cf7a4a50c9206a5d0fba8a561d363483d4f

diff --git a/components/history/core/browser/history_types.cc b/components/history/core/browser/history_types.cc
index 530831a7e968..d4af50d42322 100644
--- a/components/history/core/browser/history_types.cc
+++ b/components/history/core/browser/history_types.cc
@@ -182,11 +182,11 @@ QueryURLResult::~QueryURLResult() = default;
 
 QueryURLResult::QueryURLResult(const QueryURLResult&) = default;
 
-QueryURLResult::QueryURLResult(QueryURLResult&&) noexcept = default;
+QueryURLResult::QueryURLResult(QueryURLResult&&) = default;
 
 QueryURLResult& QueryURLResult::operator=(const QueryURLResult&) = default;
 
-QueryURLResult& QueryURLResult::operator=(QueryURLResult&&) noexcept = default;
+QueryURLResult& QueryURLResult::operator=(QueryURLResult&&) = default;
 
 // MostVisitedURL --------------------------------------------------------------
 
diff --git a/components/history/core/browser/history_types.h b/components/history/core/browser/history_types.h
index cb84ba45405f..9f2a6bcb04be 100644
--- a/components/history/core/browser/history_types.h
+++ b/components/history/core/browser/history_types.h
@@ -142,8 +142,8 @@ class QueryResults {
   QueryResults();
   ~QueryResults();
 
-  QueryResults(QueryResults&& other) noexcept;
-  QueryResults& operator=(QueryResults&& other) noexcept;
+  QueryResults(QueryResults&& other);
+  QueryResults& operator=(QueryResults&& other);
 
   void set_reached_beginning(bool reached) { reached_beginning_ = reached; }
   bool reached_beginning() { return reached_beginning_; }
-- 
2.20.1

